src/js/components/conversation.jsx

```diff
@@ -54,34 +66,146 @@ export class ConversationComponent extends Component {
 
     };
 
+    onScroll = () => {
+        const {dispatch, shouldScrollToBottom} = this.props;
+
+        // If top of Conversation component is reached, we need to fetch older messages
+        const node = findDOMNode(this);
+        if (node.scrollTop === 0) {
+            this.fetchHistory();
+        } else if (shouldScrollToBottom) {
+            // Once we've started scrolling, we don't want the default behavior to force the scroll to the bottom afterwards
+            dispatch(setShouldScrollToBottom(false));
+        }
+    };
+
+    fetchHistory = () => {
+        const {dispatch, hasMoreMessages, isFetchingMoreMessages, messages} = this.props;
+
+        const node = findDOMNode(this);
+        if (hasMoreMessages && !isFetchingMoreMessages) {
```

Comment:
Oh, yeah for sure, I'm only suggesting that the `if (hasMoreMessages && !isFetchingMoreMessages) {` statement so that it wraps the call `onScroll` makes to fetchHistory instead of doing it inside fetchHistory itself.

Also, the `const node = findDOMNode(this);` call you make isn't necessary if the if statement is false :)